src/TMxpClient.h

```diff
@@ -62,6 +66,9 @@ class TMxpClient
     virtual void setItalic(bool val) = 0;
     virtual void setUnderline(bool val) = 0;
 
+    virtual void setStyle(const QString& val) = 0;
+    virtual const QString &getStyle() = 0;
```

Comment:
Qt uses [implicit sharing](https://doc.qt.io/qt-5/implicit-sharing.html), so you don't have to return a reference - you can just return the QString fine.

Reply:
Oh, and on compiling the test code: I think you misunderstood me:
Of course I compiled and tested it locally first, but I was referring to compiling files in Mudlet/test like TEntityHandlerTest.cpp . They are not build by QCreator. This is how I overlooked the missing set/getStyle implementations in TMxpStubClient.h. In principle I'd like to resolve such issues offline b4 bothering anyone with a PR.

Reply:
Yep, remove the reference.

Running QTest - hmm, not sure how to do that from the command line :( @SlySven do you know?

Reply:
I found the old makefiles I used last time.. they need some minor modifications... working on it. I also found a test I had written for the MXPVersionTag.. not sure how it will snug into the automated macos tests.

Reply:
Actually we can see the the command it uses to run the tests, `ctest --output-on-failure` - so running that command on the CLI should do it.

Reply:
Well, but there is no ctest. It has not been installed.

Just trying to run cmake complains about CXX and tries to find cl in its place (which I understand is part of visual C++). I tried setting CXX and other env vars to gcc or c++ from mingw, but it does not work.

Then, in my archives, I found some Makefiles I used last time (I don't think I came up with them myself). They run qmake to make makefiles for each test, then make the test itself... But TLinkStoreTest references to TLuaInterpreter.o and once I add that one there are so many references I've no longer been able to find all dependencies myself.

Yes, of course, github does the compilation too.. but when I work on the test code, it is not practical to push a change, wait 15-20 minutes until it rebuilds mudlet, then returns a simply typo and then do another commit. Also in the process you see only part of the testoutput of the test programs. I really don't want to spam the mudlet repository or you guys with zillions of commits for a tiny typo.

Reply:
About making the QString & a simple QString: @vadi2, but there are a bunch of QString& just 7 lines above that one. Does it make sense to have just one function use QString and all others QString & ? On the other hand, I certainly don't feel like changing all occurences of QString & in the whole Mudlet (maybe MXP only) sources to QString.

Reply:
@vadi2 Oh, and the mac OS build now hangs at restoring ccache for ages with: 
Received 593464454 of 601853062 (98.6%), 0.6 MBs/sec
I'm not sure if it will ever succeed or give up.

I did some dummy change/commit to force a recheck.. that works, even included my VersionTagTest. Still only managed to compile a few of the tests with my separate Makefiles..

Reply:
> but there are a bunch of QString& just 7 lines above that one

Those are different, they are input parameters - that's okay. Just looking at the output param here.

Reply:
Ah, I misunderstood you! Now, that makes sense, of course. Changed it.